### ISSUE TITLE
pub publish should show same analyzer warnings pana does

### ISSUE THREAD

**Issue by eseidelGoogle** [dart-lang/pub#1739]
My understanding from @kevmoo is that warnings shown during pub publish are different from the ones which pub.dartlang.org collects as part of pana (and uses to decide the fitness of your package, etc.) . If true, it seems like we might want to make these the same?

https://github.com/dart-lang/pub-dartlang-dart/issues/562#issuecomment-343994775

**Comment by nex3** [comment]
How are these pana warnings surfaced on pub.dartlang.org? I'm not seeing anything on the [`flame` package page](https://pub.dartlang.org/packages/flame)...

**Comment by kevmoo** [comment]
They will be in the v2 design.

For now: https://analyzer-dot-dartlang-pub.appspot.com/packages/flame

**Comment by kevmoo** [comment]
@nex3 – we'll have to have a larger discussion about how to align pan and pub.

**Comment by syberside** [comment]
Hey guys, are there any updates on this ticket? 

I hit the same problem in my project: https://pub.dev/packages/activatory#-analysis-tab-
I could not understand why pub page doesn't respect my analyzer rules.

**Comment by jonasfj** [comment]
No update, we have tried to find a set of commonly accepted lints with limited luck so far :)

We are working to refactor the scoring model on pub.dev, it's not clear yet how many lints we'll have in the future.

**Comment by syberside** [comment]
@jonasfj thank you for answering. Is there any way to run the same as pub.dev analysis to validate my package locally?

**Comment by jakemac53** [comment]
You should be able to run the `pana` package locally to run the same checks

**Comment by syberside** [comment]
@jakemac53 

I tried to run pana 0.13.5 bud didn't get the same output.

```
D:\Projects\Github\Activatory>pana --source path . --scores --verbosity verbose -j
{"logName":"pana","level":"INFO","message":"Running `C:\\Program Files\\Dart\\dart-sdk\\bin\\dart.exe --version`..."}
{"logName":"pana","level":"INFO","message":"Running `flutter.bat --version --machine`..."}
CreateProcessW failed 2
{"logName":"pana.env","level":"WARNING","message":"Unable to detect Flutter version.","error":"ProcessException: Не удается найти указанный файл.\r\n\n  Command: flutter.bat --version --machine"}
{"level":"WARNING","message":"pana might update or modify files in `.`.\nAnalysis will begin in 15 seconds, hit CTRL+C to abort it.\nTo remove this message, use `--no-warning`."}
{"logName":"pana","level":"INFO","message":"Parsing pubspec.yaml..."}
{"logName":"pana","level":"INFO","message":"Running `C:\\Program Files\\Dart\\dart-sdk\\bin\\dartfmt.bat --dry-run --set-exit-if-changed D:\\Projects\\Github\\Activatory\\lib`..."}
{"logName":"pana","level":"INFO","message":"Running `C:\\Program Files\\Dart\\dart-sdk\\bin\\pub.bat upgrade --verbosity io --no-precompile`..."}
{"logName":"pana","level":"WARNING","message":"Weird: `activatory` at `file:///D:/Projects/Github/Activatory/lib/`."}
{"logName":"pana","level":"INFO","message":"Scanning direct dependencies..."}
{"logName":"pana","level":"INFO","message":"Scanning transitive dependencies..."}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support dart with runtime: native-aot because of import violation at: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_genera
tor.dart, package:activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support flutter with runtime: flutter-native because of import violation at: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value
_generator.dart, package:activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support flutter with runtime: flutter-web because of import violation at: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_ge
nerator.dart, package:activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform android because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:
activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform ios because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:acti
vatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform windows because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:
activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform linux because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:ac
tivatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform macos because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:ac
tivatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support platform web because of the import path [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:acti
vatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support dart with runtime: native-aot because of import violation at: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_genera
tor.dart, package:activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support runtime native-aot because of the import path: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, packa
ge:activatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{"logName":"pana","level":"INFO","message":"D:\\Projects\\Github\\Activatory does not support runtime web because of the import path: [package:activatory/activatory.dart, package:activatory/src/activatory.dart, package:activatory/src/value_generator.dart, package:acti
vatory/src/post-activation/reflective_fields_filler.dart, dart:mirrors]"}
{
 "runtimeInfo": {
  "panaVersion": "0.13.5",
  "sdkVersion": "2.7.1"
 },
 "packageName": "activatory",
 "packageVersion": "1.0.0",
 "pubspec": {
  "name": "activatory",
  "description": "Test data generator for Dart ecosystem. Simplifies unit testing and Test-Driven Development.",
  "version": "1.0.0",
  "homepage": "https://github.com/syberside/Activatory",
  "environment": {
   "sdk": ">=2.0.0 <3.0.0"
  },
  "dependencies": {
   "uuid": "^2.0.0"
  },
  "dev_dependencies": {
   "test": "^1.0.0",
   "mockito": "^4.0.0",
   "pedantic": "^1.9.0"
  }
 },
 "platform": {
  "components": [],
  "uses": {
   "flutter": "allowed",
   "web": "allowed",
   "other": "allowed"
  },
  "reason": "No platform restriction found in libraries."
 },
 "licenses": [
  {
   "path": "LICENSE",
   "name": "MIT"
  }
 ],
 "health": {
  "analyzeProcessFailed": false,
  "formatProcessFailed": false,
  "resolveProcessFailed": false,
  "analyzerErrorCount": 0,
  "analyzerWarningCount": 0,
  "analyzerHintCount": 0,
  "platformConflictCount": 0
 },
 "maintenance": {
  "missingChangelog": false,
  "missingExample": true,
  "missingReadme": false,
  "missingAnalysisOptions": false,
  "oldAnalysisOptions": false,
  "strongModeEnabled": true,
  "isExperimentalVersion": false,
  "isPreReleaseVersion": false
 },
 "pkgResolution": {
  "dependencies": [
   {
    "package": "charcode",
    "dependencyType": "transitive",
    "constraintType": "inherited",
    "resolved": "1.1.3"
   },
   {
    "package": "collection",
    "dependencyType": "transitive",
    "constraintType": "inherited",
    "resolved": "1.14.12"
   },
   {
    "package": "convert",
    "dependencyType": "transitive",
    "constraintType": "inherited",
    "resolved": "2.1.1"
   },
   {
    "package": "crypto",
    "dependencyType": "transitive",
    "constraintType": "inherited",
    "resolved": "2.1.4"
   },
   {
    "package": "mockito",
    "dependencyType": "dev",
    "constraintType": "normal",
    "constraint": "^4.0.0"
   },
   {
    "package": "pedantic",
    "dependencyType": "dev",
    "constraintType": "normal",
    "constraint": "^1.9.0"
   },
   {
    "package": "test",
    "dependencyType": "dev",
    "constraintType": "normal",
    "constraint": "^1.0.0"
   },
   {
    "package": "typed_data",
    "dependencyType": "transitive",
    "constraintType": "inherited",
    "resolved": "1.1.6"
   },
   {
    "package": "uuid",
    "dependencyType": "direct",
    "constraintType": "normal",
    "constraint": "^2.0.0",
    "resolved": "2.0.4"
   }
  ]
 },
 "dartFiles": {},
 "stats": {
  "analyzeProcessElapsed": 0,
  "formatProcessElapsed": 251,
  "resolveProcessElapsed": 3355,
  "totalElapsed": 9155
 },
 "tags": [
  "sdk:dart",
  "runtime:native-jit"
 ],
 "scores": {
  "health": 100.0,
  "maintenance": 100.0
 }
}
```

**Comment by sigurdm** [comment]
I don't think we want this. The validations we do on the pub-side should be "timeless" in that they cannot be updated for old client versions. The analysis options on pub.dev might change over time.

When/if we get staged publishing on pub.dev hopefully the surprises upon publishing will not be as bad.
